src/Gaffer/SubGraph.cpp

```diff
@@ -83,42 +83,58 @@ const Plug *SubGraph::correspondingInput( const Plug *output ) const
 		return nullptr;
 	}
 
+	const Plug *internalInput = nullptr;
 	if( const BoxOut *boxOut = internalOutput->parent<BoxOut>() )
 	{
-		internalOutput = boxOut->plug()->getInput();
-		if( !internalOutput )
+		if( boxOut->passThroughPlug()->getInput() )
 		{
-			return nullptr;
+			internalInput = boxOut->passThroughPlug();
+		}
+		else
+		{
+			// Prepare for legacy branch below
+			internalOutput = boxOut->plug()->getInput();
+			if( !internalOutput )
+			{
+				return nullptr;
+			}
 		}
 	}
 
-	const DependencyNode *node = IECore::runTimeCast<const DependencyNode>( internalOutput->node() );
-	if( !node )
+	if( !internalInput )
 	{
-		return nullptr;
-	}
+		// Legacy code path for networks made before BoxOut had pass-through support.
+		// These used Switch nodes wired up manually - this is the `node` referred to
```

Comment:
Just noting these are not always Switches. You can make the same work with any enabled plug on an internal node. I think the code below still works either way though, so not important really.